src/kakarot/accounts/eoa/library.cairo

```diff
@@ -122,18 +124,34 @@ namespace ExternallyOwnedAccount {
         ) = EthTransaction.decode([call_array].data_len, calldata + [call_array].data_offset);
 
         let (_kakarot_address) = kakarot_address.read();
-        // execute_at_address signature is
-        // address: felt, value: felt, gas_limit: felt, calldata_len: felt, calldata: felt*
+
         let (current_tx_calldata: felt*) = alloc();
-        assert [current_tx_calldata] = destination;
-        assert [current_tx_calldata + 1] = amount;
-        assert [current_tx_calldata + 2] = gas_limit;
-        assert [current_tx_calldata + 3] = payload_len;
-        memcpy(current_tx_calldata + 4, payload, payload_len);
+        local offset;
+        local selector;
+        // If destination is 0, we are deploying a contract
+        if (destination == FALSE) {
```

Comment:
Alternatively I could move the logic into their own functions:
```
if  (destination == FALSE) {
    deploy_contract();
}else{
    execute_contract_bytecode();
}
```
But imo this is still fine.

Reply:
still fine imo as well